core/server/api/v2/utils/serializers/input/utils/url.js

```diff
@@ -1,13 +1,26 @@
 const _ = require('lodash');
 const {absoluteToRelative, getBlogUrl, STATIC_IMAGE_URL_PREFIX} = require('../../../../../../services/url/utils');
 
+const handleCanonicalUrl = (url) => {
+    const blogDomain = getBlogUrl().replace(/^http(s?):\/\//, '').replace(/\/$/, '');
+    const absolute = url.replace(/^http(s?):\/\//, '');
+
+    if (absolute.startsWith(blogDomain)) {
+        return absoluteToRelative(url, {withoutSubdirectory: true});
```

Comment:
I have the feeling we have the check for being an external url in a couple of places in Ghost now.

Reply:
Would be cool to introduce: `urlService.utils.isExternalUrl(...)`
Def doesn't have to be part of the PR :)

Reply:
> return absoluteToRelative(url, {withoutSubdirectory: true});

References #8221. We should handle absolute -> relative on model layer, not API layer.
But the PR can stay as is for now 👍

Reply:
> We should handle absolute -> relative on model layer, not API layer.
Yes, absolutely. `absolute_url` will be included as part of the audit for https://github.com/TryGhost/Ghost/issues/10477 :ok_hand:

Reply:
> I have the feeling we have the check for being an external url in a couple of places in Ghost now.

Checked it today. This check is only performed in this input serializer, and don't think like there's enough repetition for each case to extract this specific utility function. If it needs to be used elsewhere in the codebase, then would probably make sense to extract. Leaving as is for now :+1:

Reply:
🤔 search for "external" in Ghost

e.g. core/server/services/url/utils.js
e.g. core/server/helpers/img_url.js